pkg/store/proxy_heap.go

```diff
@@ -897,11 +898,15 @@ type respSet interface {
 
 // hasInternalReplicaLabels returns true if any replica label in the series request is not an
 // external label for the given Client.
-func hasInternalReplicaLabels(st Client, req *storepb.SeriesRequest) bool {
+func hasInternalReplicaLabels(st Client, req *storepb.SeriesRequest, internalLabelResort bool) bool {
+	if !internalLabelResort {
+		return true
+	}
+
 	bloom := st.LabelNamesBloom()
 	// Empty bloom filter capacity is 1. We fallback to eager retrieval if bloom filter
 	// is yet to be updated, as we cannot yet determine if the store has internal replica labels.
-	if bloom.Cap() <= 1 {
+	if bloom == nil || bloom.Cap() <= 1 {
```

Comment:
Could `bloom` ever be nil?

Reply:
Yes, when it has yet to be initialized via the first UpdateLabelNamesBloom call. 

I noticed a panic in the test here which occurs on line 904 in https://github.com/thanos-io/thanos/pull/6317 due to a nil pointer deference: https://app.circleci.com/pipelines/github/thanos-io/thanos/13661/workflows/4023a1f2-8d19-4793-98a7-232c9df35318/jobs/23598.

Reply:
Ah, you're right, it was initialized for promMetadata but not for others. Will do that! I'd still keep this check just to prevent potential panics like this for other uses of proxy_help

Reply:
I don't know, IMO a bloom of `nil` should be considered a bug. Otherwise we have to add checks for nil all over the codebase.